apps/vaults/components/details/actions/QuickActionsButtons.tsx

```diff
@@ -0,0 +1,165 @@
+	**********************************************************************************************/
+	const	{data: vanillAllowanceFrom, isLoading: isValidatingAllowance, mutate: mutateAllowance} = useSWR(
+		canInteract && !isInputTokenEth && shouldUseVanillaAllowance ?
+		[selectedOptionFrom, spender] as TAllowanceFetcher : null,
+		retrieveAllowance,
+		{revalidateOnFocus: false}
+	);
+
+	useEffect((): void => {
+		if (shouldUseVanillaAllowance && vanillAllowanceFrom) {
+			set_allowanceFrom(vanillAllowanceFrom);
+		}
+	}, [vanillAllowanceFrom, shouldUseVanillaAllowance]);
+
+	const onSuccess = useCallback(async (): Promise<void> => {
+		onChangeAmount(DefaultTNormalizedBN);
+		await refresh();
+	}, [onChangeAmount, refresh]);
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger an approve web3 action, simply trying to approve `amount` tokens
+	** to be used by the Partner contract or the final vault, in charge of
+	** depositing the tokens.
+	** This approve can not be triggered if the wallet is not active
+	** (not connected) or if the tx is still pending.
+	**************************************************************************/
+	async function	onApproveFrom(): Promise<void> {
+		onApprove(
+			set_txStatusApprove,
+			async (): Promise<void> => {
+				if (currentSolver === Solvers.COWSWAP) {
+					set_allowanceFrom({
+						raw: ethers.constants.MaxUint256,
+						normalized: Infinity
+					});
+				} else {
+					await mutateAllowance();
+				}
+			}
+		);
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger a deposit web3 action, simply trying to deposit `amount` tokens to
+	** the selected vault.
+	**************************************************************************/
+	async function	onDepositToVault(): Promise<void> {
+		onExecuteDeposit(
+			set_txStatusDeposit,
+			async (): Promise<void> => {
+				await onSuccess();
+			}
+		);
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Trigger a withdraw web3 action using the vault contract to take back
+	** some underlying token from this specific vault.
+	**************************************************************************/
+	async function	onWithdrawShares(): Promise<void> {
+		onExecuteWithdraw(
+			set_txStatusDeposit,
+			async (): Promise<void> => {
+				await onSuccess();
+			}
+		);
+	}
+	
+	/* 🔵 - Yearn Finance ******************************************************
+	** Wrapper to decide if we should use the partner contract or not
+	**************************************************************************/
+	function	onDepositOrWithdraw(): void {
+		if (isDepositing) {
+			onDepositToVault();
+		} else {
+			onWithdrawShares();
+		}
+	}
+
+	console.warn(currentSolver);
+	if (
+		(currentSolver === Solvers.VANILLA && (isDepositing &&( txStatusApprove.pending || amount.raw.gt(allowanceFrom?.raw || 0))))
```

Comment:
We're doing a lot of `allowanceFrom?.raw || 0`, could we just make sure that `allowanceFrom.raw` if it didn't exist is 0 at this point, perhaps giving some default in the `useState()`
```typescript
const [allowanceFrom, set_allowanceFrom] = useState<TNormalizedBN>({raw: BigNumber.from(0), normalized: 0});
```

Reply:
Also, the logic condition in this `if` is a bit 🤯 :) let's try and extract this to a function and potentially simplify?